packages/ember-htmlbars/lib/hooks/invoke-helper.js

```diff
@@ -16,6 +17,29 @@ export default function invokeHelper(morph, env, scope, visitor, params, hash, h
 
   // Ember.Helper helpers are pure values, thus linkable
   if (helperStream.linkable) {
+
+    if (morph) {
+      // When processing an inline expression the params and hash have already
+      // been linked. Thus, HTMLBars will not link the returned helperStream.
+      // We subscribe the morph to the helperStream here, and also subscribe
+      // the helperStream to any params.
+      let addedDependency = false;
+      let l = params.length;
+      if (l) {
+        for (var i = 0; i < l; i++) {
```

Comment:
`let l = params.length; if(l) {` is an unnecessary optimization in modern engines. It won't hurt anything, but nor will it help. See http://jsperf.com/caching-array-length/4